host/usb/CMakeLists.txt

```
@@ -28,6 +33,23 @@ if(CONFIG_SOC_USB_OTG_SUPPORTED)
                      )
     list(APPEND include "include")
     list(APPEND priv_includes "private_include")
+
+    if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_LESS "6.0")
+        # usb_phy driver is present in USB component only in IDF 5.x
+        # If we want to override usb component from esp-idf with managed usb component from esp-usb, we must provide
+        # usb_phy driver for the managed usb component separately by taking src usb_phy.c and
+        # include esp_private/usb_phy.h from the native usb component in esp-idf
+
+        # Get path to usb component inside esp-idf
+        idf_component_get_property(esp_idf_usb_dir usb COMPONENT_OVERRIDEN_DIR)
+
+        # Explicitly add usb_phy.c and usb_phy.h from overridden usb component in esp-idf
+        list(APPEND srcs "${esp_idf_usb_dir}/usb_phy.c")
+
+        # This append must be under the first "include" append,
+        # so usb_host.h (and the rest of the files in /include/usb/) are used from esp-usb, not from esp-idf
+        list(APPEND include "${esp_idf_usb_dir}/include")
+    endif()
```

do we need to clean it up with? 

sorry, if I am missing something. 

```suggestion
    list(REMOVE_DUPLICATES srcs)
    list(REMOVE_DUPLICATES include)
    endif()
```

---

I have not used `list(REMOVE_DUPLICATES x)` beofe, I can have a look.
But the idea here is that:

`list(APPEND include "include")` this adds include directory from esp-usb's usb component
`list(APPEND include "${esp_idf_usb_dir}/include")` this adds include directory from esp-idf's usb component, due to `esp_phy.h`  located in `/component/usb/include/esp_private/esp_phy.h`. 

We can't just append include like this `list(APPEND include "${esp_idf_usb_dir}/include/esp_private")`, because for example the the include in `usb_host.c` `#include "esp_private/usb_phy.h"` would not work.

Because of that, the following:
`list(APPEND include "${esp_idf_usb_dir}/include")` must be under the `list(APPEND include "include")`, so the `usb_host.h` (and other `.h` files) is used from esp-usb's usb component, not from esp-idf's usb component.

Hope I am correct that it works like this.

---

For the completeness: `list(REMOVE_DUPLICATES include)` would not work here. 
I printed the `include` list:
`-- include list print: include;/home/peter/esp/esp-idf/components/usb/include`
One include dir is just `include`, the other one `/home/peter/esp/esp-idf/components/usb/include`

Calling `REMOVE_DUPLICATES` on `include` list would not remove anything, since there are no actual duplicates.